src/main/java/com/hierynomus/smbj/share/NamedPipe.java

```diff
@@ -141,7 +196,15 @@ public int transact(byte[] inBuffer, byte[] outBuffer) {
      * @return the number of bytes written to <code>outBuffer</code>
      */
     public int transact(byte[] inBuffer, int inOffset, int inLength, byte[] outBuffer, int outOffset, int outLength) {
-        return ioctl(FSCTL_PIPE_TRANSCEIVE, true, inBuffer, inOffset, inLength, outBuffer, outOffset, outLength);
+        if (outBuffer.length - outOffset < outLength) {
+            throw new IllegalArgumentException();
+        }
+        final byte[] transactOutBuffer = transact(inBuffer);
+        if (transactOutBuffer.length > outLength) {
+            throw new BufferOverflowException();
```

Comment:
I see a great conversation happened here but I'm not sure what the conclusion of that conversation was.  Is a change desired here?  Throw the exception, as is; or return -1?  Note that this particular set of code is not the most efficient since it means the transaction would have to be redone so the caller has to deal with further latency issues.

Another idea here is we could return a negative number that represents the amount of data missing, so for example, if you provided a 64k buffer but 70k of data was read, -4k could be returned to inform the caller an additional 4k is needed to store the data.

This call seems irrelevant to me because we are not using the length to manage the other call above i.e. it will read everything, and indeed it has to or the named pipe becomes desynchronized with the client; but it also stores everything, so as a library this method is doing no favors to the caller.

Perhaps I should break the other method out and give it a parameter that you specify the maximum amount of data you are interested in, and -1 could mean unlimited.  This solves the potential security vulnerability here, while making these other methods useful.

I don't know if you would like to see such changes and/or if you want to seem them for this PR or a future PR.  Let me know.